rules/audit-policy-content/raw.rego

```diff
@@ -0,0 +1,154 @@
+		"alertMessage": "audit policy rules do not cover key security areas or audit levels are invalid",
+		"alertScore": 5,
+		"packagename": "armo_builtins",
+		"failedPaths": [],
+		"fixPaths": [],
+		"alertObject": {"externalObjects": failed_obj},
+	}
+}
+
+# Sample rules object
+#rules:
+#  - level: RequestResponse
+#    resources:
+#    - group: ""
+#      resources: ["pods"]
+are_audit_file_rules_valid(rules) if {
+	seeked_resources_with_audit_level := {
+		"secrets": {
+			"auditLevel": "Metadata",
+			"mode": "equal",
+		},
+		"configmaps": {
+			"auditLevel": "Metadata",
+			"mode": "equal",
+		},
+		"tokenreviews": {
+			"auditLevel": "Metadata",
+			"mode": "equal",
+		},
+		"pods": {
+			"auditLevel": "None",
+			"mode": "not-equal",
+		},
+		"deployments": {
+			"auditLevel": "None",
+			"mode": "not-equal",
+		},
+		"pods/exec": {
+			"auditLevel": "None",
+			"mode": "not-equal",
+		},
+		"pods/portforward": {
+			"auditLevel": "None",
+			"mode": "not-equal",
+		},
+		"pods/proxy": {
+			"auditLevel": "None",
+			"mode": "not-equal",
+		},
+		"services/proxy": {
+			"auditLevel": "None",
+			"mode": "not-equal",
+		},
+	}
+
+	# Policy file must contain every resource
+	some resource, config in seeked_resources_with_audit_level
+
+	# Every seeked resource mu have valid audit levels 
+	not test_all_rules_against_one_seeked_resource(resource, config, rules)
+}
+
+test_all_rules_against_one_seeked_resource(seeked_resource, value_of_seeked_resource, rules) if {
+	# Filter down rules to only those concerning a seeked resource
+	rules_with_seeked_resource := [rule | rule := rules[_]; is_rule_concering_seeked_resource(rule, seeked_resource)]
+	rules_count := count(rules_with_seeked_resource)
+
+	# Move forward only if there are some
+	rules_count > 0
+
+	# Check if rules concerning seeked resource have valid audit levels 
+	valid_rules := [rule | rule := rules_with_seeked_resource[_]; test_one_rule_against_seeked_resource_with_proper_audit_level(rule, seeked_resource, value_of_seeked_resource)]
+	valid_rules_count := count(valid_rules)
+
+	valid_rules_count > 0
+
+	# Compare all rules for that specififc resource with those with valid rules, if amount of them differs,
+	# it means that there are also some rules which invalid audit level
+	valid_rules_count == rules_count
+}
+
+is_rule_concering_seeked_resource(rule, seeked_resource) if {
+	rule.resources[_].resources[_] == seeked_resource
+}
+
+# Sample single rule:
+#  	 level: RequestResponse
+#    resources:
+#    - group: ""
+#      resources: ["pods"]
+test_one_rule_against_seeked_resource_with_proper_audit_level(rule, seeked_resource, value_of_seeked_resource) if {
+	validate_rule_audit_level(rule, value_of_seeked_resource)
+}
+
+validate_rule_audit_level(rule, value_of_seeked_resource) := result if {
+	value_of_seeked_resource.mode == "equal"
+	result := rule.level == value_of_seeked_resource.auditLevel
+} else := result {
+	result := rule.level != value_of_seeked_resource.auditLevel
```

Comment:
```suggestion
	rule.level != value_of_seeked_resource.auditLevel
```

Reply:
indeed, result obj is not needed

Reply:
when i remove it, tests stops passing but at first glance, seems to be the same..